src/plugin/persist.js

```diff
@@ -239,7 +239,10 @@ eg.module("persist", ["jQuery", window, document], function($, global, doc) {
 			key = GLOBAL_KEY;
 			data = arguments.length === 1 ? state : null;
 		}
-		data && setStateByKey(key, data);
+
+		if (data || arguments.length === 2) {
+			setStateByKey(key, data);
+		}
```

Comment:
IMO, thre's no need check arguments.length. Just receiving as param is more clear enough.

```js
$.persist = function(state, data) {
	var key;

	if (typeof state === "string") {
		key = state;
	} else {
		key = GLOBAL_KEY;
		data = arguments.length === 1 ? state : null;
	}

	data && setStateByKey(key, data);

	return getStateByKey(key);
};
```